main.rb

```diff
@@ -0,0 +1,186 @@
+    return enum_for(:my_each_with_index) unless block_given?
+
+    new_self = self
+    i = 0
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    new_self.length.times do
+      yield(new_self[i], i)
+      i += 1
+    end
+    new_self
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    i = 0
+    arr = []
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    new_self.length.times do
+      yield(new_self[i]) == true ? arr = arr.push(new_self[i]) : arr
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check == new_self.length
+  end
+
+  def my_any?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? true : false
+  end
+
+  def my_none?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? false : true
+  end
+
+  def my_count
+    return enum_for(:my_count) unless block_given?
+
+    self_length = length
+    i = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    length.times do
+      yield(new_self[i]) == false ? self_length -= 1 : self_length = self_length
+      i += 1
+    end
+    self_length
+  end
```

Comment:
# my_count
- [ ] `my_count` returns the number of items in enum through enumeration

      Failure/Error: expect(array.my_count).to eq array.count

        expected: 100
             got: #<Enumerator: [8, 2, 4, 3, 2, 1, 3, 7, 6, 6, 4, 8, 7, 0, 2, 5, 4, 2, 7, 5, 2, 0, 6, 8, 6, 0, 8, 4, 6,... 0, 1, 8, 6, 0, 7, 6, 1, 8, 6, 8, 0, 0, 3, 4, 4, 4, 0, 4, 1, 5, 2, 2, 8, 6, 3, 5, 2, 8, 8]:my_count>

##    
- [ ] `my_count` returns the number of items in enum through enumeration::range

      Failure/Error: expect(range.my_count).to eq range.count

        expected: 46
             got: #<Enumerator: 5..50:my_count>
##
- [ ] `my_count` counts the number of items in enum that are equal to item if an argument is given

      Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)

      ArgumentError:
        wrong number of arguments (given 1, expected 0)